rosrust/src/api/ros.rs

```diff
@@ -391,7 +391,20 @@ impl Ros {
     }
 
     pub(crate) fn subscribe_param<'a, T: Deserialize<'a>>(&self, key : &str, callback: Arc<dyn Fn()->() + Send + Sync>) -> Result<()> {
-        self.master.subscribe_param::<T>(key)?;
+        let res = self.master.subscribe_param::<T>(key);
+        match res {
+            Ok(_) => {},
+            Err(ResponseError::Server(ref reason)) => {
+                if reason.starts_with("Response data has unexpected structure") {
+                    // this case is fine, we probably subscribed to a float and it's set to int or something
```

Comment:
what happens if the param is set to a string but expects a float, or the other way around?